realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1657,6 +1657,91 @@ public void distinctAsync_notIndexedLinkedFields() {
         }
     }
 
+    /*
+     * The idea is to interweave different values in 2's multiplier and 3's multiplier in a way that
+     * the outcome is different if the order of distinct* operations alternates.
+     * An example is illustrated below.
+     *
+     * Object        : O1| O2| O3| O4| O5| O6
+     * indexedString : A | A | B | B | A | A
+     * indexedLong   : 1 | 1 | 1 | 2 | 2 | 2
+     *
+     * Suppose we're to execute distinct("indexedString").distinct("indexedLong"), the outcome
+     * should be O1(A, 1) and total count is 1 since distinct takes the first element when two
+     * elements are identical.
+     *
+     * In case of distinct("indexedLong").distinct("indexedString"), the outcome should be
+     * O1(A,1) and O2(B,2) so total count is 2.
+     *
+     */
+    private void populateForChainedOperation(Realm realm, long numberOfBlocks) {
+        realm.beginTransaction();
+        for (int i = 0; i < numberOfBlocks; i++) {
+            for (int j = 0; j < 6; j++) {
+                AnnotationIndexTypes obj = realm.createObject(AnnotationIndexTypes.class);
+                obj.setIndexString((((j / 2) % 2) == 0) ? "A" : "B");
+                obj.setIndexLong((j < 3) ? 1 : 2);
+            }
+        }
+        realm.commitTransaction();
+    }
+
+    @Test
+    public void distinct_checkChainedOpertionValidity() {
+        final long numberOfBlocks = 1; // must be greater than 1
+        populateForChainedOperation(realm, numberOfBlocks);
+
+        final RealmResults<AnnotationIndexTypes> distinctStringAndLong = realm.where(AnnotationIndexTypes.class).findAll().distinct("indexString").distinct("indexLong");
+        final RealmResults<AnnotationIndexTypes> distinctLongAndString = realm.where(AnnotationIndexTypes.class).findAll().distinct("indexLong").distinct("indexString");
+        assertEquals("distinctStringAndLong", 1, distinctStringAndLong.size());
```

Comment:
In terms of API representation, these two lines flow as
`Table.where().findAll()` ->  `TableView.distinct()` ->  `TableView.distinct()` (two `TableView` instances are the same).